### PR TITLE
fakeLaser with map option

### PR DESCRIPTION
**Pull request by randaz81**
added option `--test map_test` to load a map inside `FakeLaser `and simulate obstacle detection.

### CODE REVIEW

File: src/libYARP_OS/include/yarp/os/impl/SemaphoreImpl.h [hunk 1]
```diff
@@ -20,7 +20,8 @@ namespace yarp {
 //Decide which implementation to use
 #include <yarp/conf/system.h>
 
-#ifdef YARP_HAS_CXX11
+#if 0
```
Comment (drdanz) [inline src/libYARP_OS/include/yarp/os/impl/SemaphoreImpl.h]: Please remove this change as it is not related to this patch

File: src/devices/fakeLaser/CMakeLists.txt [hunk 2]
```diff
@@ -20,7 +20,8 @@ if(NOT SKIP_fakeLaser)
                             fakeLaser.cpp)
   target_link_libraries(fakeLaser YARP_OS
                                   YARP_sig
-                                  YARP_dev)
+                                  YARP_dev
+                                  YARP_math)
```
Comment (drdanz) [inline src/devices/fakeLaser/CMakeLists.txt]: You need to add a `get_property` a few lines before to find YARP_math headers

Comment (randaz81) [inline src/devices/fakeLaser/CMakeLists.txt]: done
